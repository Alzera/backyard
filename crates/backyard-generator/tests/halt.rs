use backyard_generator::generate;
use backyard_parser::arena_parse;

#[test]
fn basic() {
  let arena = bumpalo::Bump::new();
  let asts = arena_parse(
    &arena,
    true,
    "$a = false;\n__halt_compiler();\nThis text is invalid"
  ).unwrap();
  insta::assert_yaml_snapshot!(generate(&asts).unwrap());
}
