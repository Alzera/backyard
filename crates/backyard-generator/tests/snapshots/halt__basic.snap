---
source: crates/backyard-generator/tests/halt.rs
assertion_line: 8
expression: generate(&asts).unwrap()
snapshot_kind: text
---
"$a = false;\n__halt_compiler();;"
